frontend/src/components/Overlay/OverlayHeader.test.tsx

```diff
@@ -0,0 +1,24 @@
+import { render, screen } from '@testing-library/react';
+import { describe, expect, test, vi } from 'vitest';
+
+import OverlayHeader from './OverlayHeader';
+
+const mockCloseOverlay = vi.fn();
+
+describe('Overlay Header component', () => {
+	test('renders the heading text passed in', () => {
+		render(
+			<OverlayHeader heading="Test Header" closeOverlay={mockCloseOverlay} />
+		);
+
+		expect(screen.getByText('Test Header')).toBeInTheDocument();
+	});
+
+	test('renders close button', () => {
+		render(
+			<OverlayHeader heading="Test Header" closeOverlay={mockCloseOverlay} />
+		);
+
+		expect(screen.getByText('close')).toBeInTheDocument();
+	});
```

Comment:
There's one more test you could add for the sake of completion: when you click the close button, the callback is fired.

For that you'll need user-event: see LevelSelectionBox for an example, though note you'll only need it for this new test, so you can use a raw `await userEvent.click()` within the test itself, without needing to do `userEvent.setup()`. Ask if you're unsure.